version: '3.7'

services:

  back:
    container_name: "baches_app"
    build:
      context: ./src
      dockerfile: Dockerfile
    working_dir: /code/back
    depends_on:
      - db
    volumes:
      - ./src:/code/back
      - static:/static
    environment:
      - POSTGRES_ENGINE=django.contrib.gis.db.backends.postgis
      - POSTGRES_DB=realstate
      - POSTGRES_USER=hello_django
      - POSTGRES_PASSWORD=hello_django
      - POSTGRES_HOST=db
      - POSTGRES_PORT=5433


      # Para pruebas.
      - DJANGO_SUPERUSER_USERNAME=admin
      - DJANGO_SUPERUSER_EMAIL=admin@example.com
      - DJANGO_SUPERUSER_PASSWORD=adminpass123

    ports:
      - "8000:8000"   
    restart: on-failure
    networks:
      - gis_network

  ngnx:
    container_name: "baches_nginx"
    build: ./nginx
    volumes:
      - static:/static
    ports:
      - "80:80"
    depends_on:
      - back
    networks:
      - gis_network
    restart: on-failure

  db:
    container_name: "baches_db"
    image: "postgis/postgis"
    restart: always
    expose:
      - "5433"
    ports:
      - '5433:5433'
    command: -p 5433
    volumes:
      - db_persdata:/var/lib/postgrespers
    environment:
      - POSTGRES_ENGINE=django.db.backends.postgresql
      - POSTGRES_DB=realstate
      - POSTGRES_USER=hello_django
      - POSTGRES_PASSWORD=hello_django
      - POSTGRES_HOST=db
      - POSTGRES_PORT=5433
    networks:
      - gis_network

networks:
  gis_network:
    driver: bridge

volumes:
  db_persdata : {}
  static:


